File: mvvm-java/domain/src/test/java/com/wasisto/githubuserfinder/domain/usecases/GetUserUseCaseTest.java
package com.wasisto.githubuserfinder.domain.usecases;

import com.wasisto.githubuserfinder.domain.interfaces.GithubDataSource;
import com.wasisto.githubuserfinder.domain.interfaces.SearchHistoryDataSource;
import com.wasisto.githubuserfinder.domain.models.User;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import static org.junit.Assert.*;
import static org.mockito.Mockito.when;

public class GetUserUseCaseTest {

    private GetUserUseCase getUserUseCase;

    @Mock
    private GithubDataSource githubDataSource;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
        getUserUseCase = new GetUserUseCase(githubDataSource);
    }

    @Test
    public void testExecute() throws Exception {
        String username = "octocat";
        User user = new User() {{
            setLogin(username);
            setAvatarUrl("https://avatars.githubusercontent.com/u/583231?v=4");
            setBlog("https://github.blog");
        }};

        when(githubDataSource.getUser(username)).thenReturn(user);

        assertEquals(user, getUserUseCase.execute(username));
    }
}

File: mvp-kotlin/domain/src/test/java/com/wasisto/githubuserfinder/domain/usecases/GetUserUseCaseTest.kt
package com.wasisto.githubuserfinder.domain.usecases

import com.wasisto.githubuserfinder.domain.interfaces.GithubDataSource
import com.wasisto.githubuserfinder.domain.models.User
import io.mockk.MockKAnnotations
import io.mockk.every
import io.mockk.impl.annotations.MockK
import org.junit.Before
import org.junit.Test

import org.junit.Assert.*

class GetUserUseCaseTest {

    private lateinit var getUserUseCase: GetUserUseCase

    @MockK
    private lateinit var githubDataSource: GithubDataSource

    @Before
    fun setUp() {
        MockKAnnotations.init(this, relaxUnitFun = true)
        getUserUseCase = GetUserUseCase(githubDataSource)
    }

    @Test
    fun testInvoke() {
        val username = "octocat"
        val user = User(
            login = username,
            avatarUrl = "https://avatars.githubusercontent.com/u/583231?v=4",
            blog = "https://github.blog"
        )

        every { githubDataSource.getUser(username) }.returns(user)

        assertEquals(user, getUserUseCase(username))
    }
}